web_services/profile/app.js

```diff
@@ -33,149 +35,209 @@ app.post("/upload/:channel", function (req, res) {
     res.status(400).json({ error: "missing uuid" });
     return;
   }
+  // Don't allow too high TTL values
+  if (requestedTtl && requestedTtl > config.stdTTL) {
+    res.status(400).json({ error: "requested TTL too high" });
+    return;
+  }
 
-  let cacheEntry
-  const profile = {uuid, data}
-
-  const current_data = dataCache.get(channel)
-  if (current_data) {
-    // Check if there is already a profile with the provided uuid to prevent duplicates
-    const existingProfile = current_data.find(entry => (entry.uuid === uuid))
-    if (existingProfile) {
-      if (existingProfile.data === profile.data) {
-        console.log(`Received duplicate profile ${uuid} for channel ${channel}`)
-        // restart TTL counter of channel
-        dataCache.ttl(channel)
-        res.status(201).json({ success: true });
-      } else {
-        // Same UUID but different content? This is scary. Likely client bug. Bail out.
-        res.status(500).json({error: `Profile ${uuid} already exists in channel ${channel} with different data.`});
-      }
-      return;
-    }
-
-    const isRecoveryChannel = (uuid) =>
-        uuid.startsWith('connection_') ||
-        uuid.startsWith('group_') ||
-        uuid.startsWith('sig_');
+  // use standard TTL if nothing provided by client
+  const ttl = requestedTtl || config.stdTTL
 
-    // Bail out if channel is full but ignore the limit if it's a recovery channel
-    if (current_data.length >= config.channel_entry_limit && !isRecoveryChannel(uuid)) {
-      res.status(config.channel_limit_response_code).json({error: config.channel_limit_message});
-      return;
+  let channel = channelCache.get(channelId)
+  if (!channel) {
+    // Create new channel.
+    channel = {
+      cache: new NodeCache(config.channel_entry_config),
```

Comment:
Is there any documentation about the possibility of using a `NodeCache` object as the value of another NodeCache object?

Reply:
If the lib is not serializing the value and is storing them as original js object, why should not simply use a dict/object for storing channel entries instead of a new NodeCache object when we do not need expiration time?